packages/voter-dapp/src/RetrieveRewards.js

```diff
@@ -205,15 +207,32 @@ function RetrieveRewards({ votingAccount }) {
   // Construct the claim rewards transaction.
   const rewardsTxn = useRetrieveRewardsTxn(retrievedRewardsEvents, reveals, votingAccount);
 
+  // Check `retrieveRewards` return value without submitting the transaction, and save it to state.
+  useEffect(() => {
+    const VotingContract = drizzle.contracts.Voting;
+    if (rewardsTxn.oldestUnclaimedRound && rewardsTxn.toRetrieve) {
+      VotingContract.methods
+        .retrieveRewards(votingAccount, rewardsTxn.oldestUnclaimedRound.toString(), rewardsTxn.toRetrieve)
+        .call()
+        .then(_pendingRewards => {
+          setPendingRewards(web3.utils.fromWei(_pendingRewards.toString()));
+        })
+        .catch(err => console.error(`retrieveRewards.call failed:`, err));
+    }
+  }, [votingAccount, rewardsTxn.oldestUnclaimedRound, rewardsTxn.toRetrieve]);
+
   let body = "";
   const hasPendingTxns = rewardsTxn.status === "pending";
   if (!rewardsTxn.ready) {
     body = "Loading";
   } else if (rewardsTxn.send) {
     body = (
-      <Button onClick={rewardsTxn.send} variant="contained" color="primary" disabled={hasPendingTxns}>
-        Claim Your Rewards
-      </Button>
+      <>
+        <div>Claimable rewards: {pendingRewards}</div>
```

Comment:
Could you wrap this in `formatWithMaxDecimals` so that it nicely comma separates and delimits the number of rewards?

Reply:
Done

Reply:
![Screen Shot 2020-10-06 at 09 04 16](https://user-images.githubusercontent.com/9457025/95205095-eb688200-07b2-11eb-9f76-30340eb62060.png)